KeyVaultSampleBase.cs

```diff
@@ -55,30 +62,39 @@ public KeyVaultSampleBase()
+            var clientSecret = ConfigurationManager.AppSettings[SampleConstants.ConfigKeys.SPSecret];
+            var clientId = ConfigurationManager.AppSettings[SampleConstants.ConfigKeys.ApplicationId];
             var subscriptionId = ConfigurationManager.AppSettings[SampleConstants.ConfigKeys.SubscriptionId];
             var resourceGroupName = ConfigurationManager.AppSettings[SampleConstants.ConfigKeys.ResourceGroupName];
             var vaultLocation = ConfigurationManager.AppSettings[SampleConstants.ConfigKeys.VaultLocation];
             var vaultName = ConfigurationManager.AppSettings[SampleConstants.ConfigKeys.VaultName];
+            var azureEnvironment = ConfigurationManager.AppSettings[SampleConstants.ConfigKeys.AzureEnvironment];
 
-            InstantiateSample(tenantId, spObjectId, appId, spSecret, subscriptionId, resourceGroupName, vaultLocation, vaultName);
+            InstantiateSample(tenantId, clientSecret, clientId, objectId, subscriptionId, resourceGroupName, vaultLocation, vaultName, azureEnvironment);
         }
 
-        private void InstantiateSample(string tenantId, string objectId, string appId, string appSecret, string subscriptionId, string resourceGroupName, string vaultLocation, string vaultName)
-        {
-            context = ClientContext.Build(tenantId, objectId, appId, subscriptionId, resourceGroupName, vaultLocation, vaultName);
+        private void InstantiateSample(string tenantId, string clientSecret, string clientId, string objectId, string subscriptionId, string resourceGroupName, string vaultLocation, string vaultName, string azureEnvironment)
+        {           
+            context = ClientContext.Build(tenantId, clientSecret, clientId, objectId, subscriptionId, resourceGroupName, vaultLocation, vaultName);
 
+            var environment = context.Environment(azureEnvironment);
             // log in with as the specified service principal
-            var serviceCredentials = Task.Run(() => ClientContext.GetServiceCredentialsAsync(tenantId, appId, appSecret)).ConfigureAwait(false).GetAwaiter().GetResult();
+            var credential = Task.Run(() => context.GetAzureCredentialsAsync(clientId, clientSecret, tenantId, environment)).ConfigureAwait(false).GetAwaiter().GetResult();
+            
+            var restClient = RestClient.Configure()
+                .WithEnvironment(credential.Environment)
+                .WithCredentials(credential)
+                .Build();
 
-            // instantiate the management client
-            ManagementClient = new KeyVaultManagementClient(serviceCredentials);
+            ManagementClient = new KeyVaultManagementClient(restClient);
             ManagementClient.SubscriptionId = subscriptionId;
 
-            // instantiate the data client
-            DataClient = new KeyVaultClient(ClientContext.AcquireTokenAsync);
+            //Get current Authority Host for Azure
+            TokenCredentialOptions options = new TokenCredentialOptions();
```

Comment:
if we did not set options.AuthorityHost, ClientSecretCredential will always set as default "AzureGlobalCloud"